### PR TITLE
Streamline defaulted value of ETCDConfig

### LINKED ISSUE #5157: Inconsistent default values for ETCDConfig
**How to categorize this issue?**

/area control-plane
/kind technical-debt

**What would you like to be added**:
The default values for [ETCDConfig](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/apis/config/v1alpha1/types.go) in the `GardenletConfiguration` should be consistent which is currently not the case for `worker` configurations. For instance, if `ETCDConfig == nil` then only the `ETCDController` worker configuration is set ([ref](https://github.com/gardener/gardener/blob/e59f4f3e1f72b5eb27329cb71c7e64a89cddfa86/pkg/operation/botanist/component/etcd/bootstrap.go#L325-L328)) and the remaining default values will be determined by [gardener/etcd-druid](https://github.com/gardener/etcd-druid), e.g. `custodian-workers` will be `3`. 
However, if a `ETCDConfig` is given but `CustodianController == nil` then `custodian-workers` will be `10`.

This handling should be made consistently, i.e. `custodian-workers` should always be defaulted to `10`.

Defaulting should therefore happen in https://github.com/gardener/gardener/blob/master/pkg/gardenlet/apis/config/v1alpha1/defaults.go like done for other configs and not in https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/component/etcd/bootstrap.go.

Thanks @ialidzhikov for reporting this.

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area control-plane
/kind technical-debt

**What this PR does / why we need it**:
Previously ETCDConfig values were defaulted at different places like [here](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/component/etcd/bootstrap.go). Which caused inconsistencies. This PR make changes so the ETCDConfig values should be defaulted only [here](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/apis/config/v1alpha1/defaults.go). Which is the desired place to default all the  `GardenletConfiguration` values.
 
**Which issue(s) this PR fixes**:
Fixes #5157

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/gardenlet/apis/config/v1alpha1/defaults.go [hunk 1]
```diff
@@ -480,3 +484,31 @@ func SetDefaults_Logging(obj *Logging) {
 		obj.Loki.Garden.Storage = &DefaultCentralLokiStorage
 	}
 }
+
+// SetDefaults_ETCDConfig sets defaults for the ETCD.
+func SetDefaults_ETCDConfig(obj *ETCDConfig) {
```
Comment (krgostev) [inline pkg/gardenlet/apis/config/v1alpha1/defaults.go:489]: Could you add unit test [here](https://github.com/gardener/gardener/blob/master/pkg/gardenlet/apis/config/v1alpha1/defaults_test.go) for this function?

**Review by krgostev — CHANGES_REQUESTED** [review]

**Comment by rfranzke** [comment]
Can you please add a proper PR description?

**Review by abdasgupta — APPROVED** [review]
Looks fine to me

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**